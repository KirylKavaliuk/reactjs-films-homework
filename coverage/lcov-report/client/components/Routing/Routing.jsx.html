<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/Routing/Routing.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/Routing</a> Routing.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import {
  Switch,
  Route,
} from 'react-router-dom';
&nbsp;
import MoviesList from 'containers/MoviesList/MoviesList';
&nbsp;
import NotFound from 'components/NotFound/NotFound';
&nbsp;
const pathes = [{
  exact: true,
  value: '/',
}, {
  exact: true,
  value: '/trading',
}, {
  exact: true,
  value: '/top-rated',
}, {
  exact: true,
  value: '/coming-soon',
}, {
  exact: true,
  value: '/search',
}, {
  exact: true,
  value: '/genre/:genreId',
}];
&nbsp;
const Routing = () =&gt; (
  &lt;Switch&gt;
    { pathes.map((path, index) =&gt; (
      &lt;Route key ={ index } exact={ path.exact } path={ path.value } component={ MoviesList }/&gt;
    )) }
    &lt;Route component={ NotFound }/&gt;
  &lt;/Switch&gt;
);
&nbsp;
export default Routing;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
