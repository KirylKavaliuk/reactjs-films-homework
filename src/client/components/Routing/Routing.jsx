import React from 'react';
import {
  Switch,
  Route,
} from 'react-router-dom';

import MoviesList from 'containers/MoviesList/MoviesList';

import NotFound from 'components/NotFound/NotFound';

const pathes = [{
  exact: true,
  value: '/',
}, {
  exact: true,
  value: '/trading',
}, {
  exact: true,
  value: '/top-rated',
}, {
  exact: true,
  value: '/coming-soon',
}, {
  exact: true,
  value: '/search',
}, {
  exact: true,
  value: '/genre/:genreId',
}];

const Routing = () => (
  <Switch>
    { pathes.map((path, index) => (
      <Route
        key ={ index }
        exact={ path.exact }
        path={ path.value }
        component={ MoviesList }
      />
    )) }
    <Route component={ NotFound }/>
  </Switch>
);

export default Routing;
