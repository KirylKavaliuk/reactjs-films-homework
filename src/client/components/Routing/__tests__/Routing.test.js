import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Routing from '../Routing';

it('<Routing/>', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Routing/>);

  const tree = renderer.getRenderOutput();

  expect(tree).toMatchSnapshot();
});
