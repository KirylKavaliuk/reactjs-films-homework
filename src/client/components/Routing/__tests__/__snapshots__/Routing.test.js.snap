// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routing/> 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/trading"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/top-rated"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/coming-soon"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/search"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/genre/:genreId"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
